internal/extender/resource.go

```diff
@@ -336,92 +337,66 @@ func (s *SparkSchedulerExtender) potentialNodes(availableNodesSchedulingMetadata
 }
 
 func (s *SparkSchedulerExtender) selectExecutorNode(ctx context.Context, executor *v1.Pod, nodeNames []string) (string, string, error) {
-	resourceReservation, ok := s.resourceReservations.Get(executor.Namespace, executor.Labels[common.SparkAppIDLabel])
-	if !ok {
-		return "", failureInternal, werror.Error("failed to get resource reservations")
+	alreadyBoundNode, err := s.resourceReservationManager.FindAlreadyBoundReservationNode(ctx, executor)
+	if err != nil {
+		return "", failureInternal, werror.WrapWithContextParams(ctx, err, "error when looking for already bound reservations")
 	}
-	unboundReservations, outcome, unboundResErr := s.findUnboundReservations(ctx, executor, resourceReservation)
-	if unboundResErr != nil {
-		extraExecutorCount := 0
-		if sr, ok := s.softReservationStore.GetSoftReservation(executor.Labels[common.SparkAppIDLabel]); ok {
-			extraExecutorCount = len(sr.Reservations)
-		}
-		driver, err := s.podLister.getDriverPod(ctx, executor)
-		if err != nil {
-			return "", failureInternal, err
+	if alreadyBoundNode != "" {
+		// check that it is part of the nodes passed and return it
+		if resultNode, ok := s.getReservationNodeFromNodeList([]string{alreadyBoundNode}, nodeNames); ok {
```

Comment:
in the case where this condition is false, what happens?

Reply:
If this is false, that means that we have a previously bound node for this executor, but we did not get it from `kube-scheduler` as a possible node (because it could no longer be available for example), so we move on and try to find other unbound reservations to take.